Assets/Scripts/Models/Job/RequestedItem.cs

```diff
@@ -0,0 +1,84 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+
+using MoonSharp.Interpreter;
+using UnityEngine;
+
+namespace ProjectPorcupine.Jobs
+{
+    [MoonSharpUserData]
+    public class RequestedItem
+    {
+        public RequestedItem(string type, int minAmountRequested, int maxAmountRequested)
+        {
+            Type = type;
+            MinAmountRequested = minAmountRequested;
+            MaxAmountRequested = maxAmountRequested;
+        }
+
+        public RequestedItem(string type, int maxAmountRequested)
+            : this(type, maxAmountRequested, maxAmountRequested)
+        {
+        }
+
+        public RequestedItem(RequestedItem item)
+            : this(item.Type, item.MinAmountRequested, item.MaxAmountRequested)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProjectPorcupine.Job.RequestedItem"/> class with the amount needed to fill the inventory item passed in.
+        /// </summary>
+        public RequestedItem(Inventory inventory)
+            : this(inventory.Type, inventory.MaxStackSize - inventory.StackSize)
+        {
+        }
+
+        public string Type { get; set; }
+
+        public int MinAmountRequested { get; set; }
+
+        public int MaxAmountRequested { get; set; }
+
+        public RequestedItem Clone()
+        {
+            return new RequestedItem(this);
+        }
+
+        public bool NeedsMore(Inventory inventory)
+        {
+            return AmountNeeded(inventory) > 0;
+        }
+
+        public bool DesiresMore(Inventory inventory)
+        {
+            return AmountDesired(inventory) > 0;
+        }
+
+        public int AmountDesired(Inventory inventory = null)
+        {
+            if (inventory == null || inventory.Type != Type)
+            {
+                return MaxAmountRequested;
```

Comment:
@mikejbrown @TomMalbran I'm open to better names on the variables and methods.

I understand that passing in an irrelevant type seems confusing. The way I was thinking about the request you are pointing out is that you aren't asking how many more items similar to what the character carries that the job wants. Instead you are asking a specific requested material, like steel plate, how many are desired. Passing in the inventory allows it to account for what the character is carrying.

If the job wants 20 steel plate and the character is carrying iron ore, the desired amount of steel plate is still 20. If the character is carrying 10 steel plate, then the desired amount is 10.

@TomMalbran Is correct about what the variables are representing. If the min and max are set to 5 and 50 respectively, it needs 5 but desires 50. But as I said. I'm not locked in on the names. I'm going to mark this as `WIP` until we figure out a way to clarify this.